### PR TITLE
Move eval functions to pyodide.code submodule

### PR DESCRIPTION
**Pull request by hoodmane**
This raises a deprecation warning the first time such a function is used. Work towards #2689.

### CODE REVIEW

File: src/py/pyodide/__init__.py [hunk 1]
```diff
@@ -64,20 +75,32 @@
...
+
+def __getattr__(name):
+    if name in DEPRECATED_LIST:
+        warn(
+            f"{name} has been moved to pyodide.{DEPRECATED_LIST[name]}. "
+            "Accessing it through the pyodide module is deprecated.",
+            DeprecationWarning,
+        )
+        globals()[name] = globals()[f"_deprecated_{name}"]
+        return globals()[name]
```
Comment (rth) [inline src/py/pyodide/__init__.py:105]: Do you need to rename at all:  I would have thought that just returning `globals()[name]` with a warning would be enough?  Though yes, they seem to be doing it in https://peps.python.org/pep-0562/

**Review by rth — COMMENTED** [review]
+1 for creating a separate module with those, but I think we might want to avoid overwriting builtin name. Otherwise,
```
from pyodide import eval
```
will lead to confusing errors for users.

Maybe `pyodide.code`?

File: src/py/pyodide/__init__.py (hunk 1 above)
Comment (hoodmane) [reply src/py/pyodide/__init__.py:105]: This prevents the warning from being raised more than once per input.

File: pyodide-test-runner/pyodide_test_runner/tests/test_decorator.py [hunk 2]
```diff
@@ -250,7 +250,7 @@ def test_raises_jsexception(selenium):
 
     @run_in_pyodide
     def raise_jsexception(selenium):
-        from pyodide import run_js
```
Comment (rth) [inline pyodide-test-runner/pyodide_test_runner/tests/test_decorator.py:253]: Also needs renaming (here and above).

**Review by rth — APPROVED** [review]
Thanks! LGTM, once CI passes.

**Pull request merged**